pkg/utils/value/value_test.go

```diff
@@ -0,0 +1,182 @@
+package value
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const beginFlag = "beginFlag"
+
+var (
+	val = map[string]interface{}{
+		beginFlag: "flag",
+		"string":  "b",
+		"int":     1,
+		"float64": 1.2,
+		"map": map[string]interface{}{
+			"internal": "internal",
+		},
+	}
+)
+
+func TestValue_Walk(t *testing.T) {
+	var logs []interface{}
+	type args struct {
+		callback func(m MapValue, k string, v interface{}, e Extra) error
+	}
+	type TestCase struct {
+		name    string
+		val     MapValue
+		args    args
+		wantErr assert.ErrorAssertionFunc
+		assert  func(t *testing.T, tt TestCase)
+	}
+
+	tests := []TestCase{
+		{
+			name: "walk",
+			val:  val,
+			args: args{
+				callback: func(m MapValue, k string, v interface{}, e Extra) error {
+					logs = append(logs, k)
+					logs = append(logs, v)
+					return nil
+				},
+			},
+			wantErr: assert.NoError,
+
+			assert: func(t *testing.T, tt TestCase) {
+				// fix map order
+				logs = fixMapOrder(logs)
+				wantLogs := []interface{}{beginFlag, "flag", "string", "b", "int", 1,
+					"float64", 1.2, "internal", "internal"}
+				assert.Equal(t, wantLogs, logs)
+			},
+		},
+		{
+			name: "break",
+			val:  val,
+			args: args{
+				callback: func(m MapValue, k string, v interface{}, e Extra) error {
+					logs = append(logs, k)
+					logs = append(logs, v)
+					if k == "string" {
+						return errors.New("break")
+					}
+					return nil
+				},
+			},
+			wantErr: assert.Error,
+			assert: func(t *testing.T, tt TestCase) {
+				assert.Equal(t, "string", logs[len(logs)-2])
+				assert.Equal(t, "b", logs[len(logs)-1])
+			},
+		},
+		{
+			name: "",
```

Comment:
maybe we could merge the case of path to other test cases?

Reply:
If combined, the test will be messy